flake.nix

```
@@ -0,0 +1,134 @@
+
+              artery-font-format = pkgs.callPackage ./nix/artery-font-format.nix { };
+
+              polyvox = pkgs.callPackage ./nix/polyvox.nix { };
+
+              gif_creator = pkgs.callPackage ./nix/gif_creator.nix { };
+
+              scribe = pkgs.callPackage ./nix/scribe.nix { };
+
+              overte = pkgs.callPackage ./nix/overte.nix {
+                inherit (self'.packages)
+                  glad
+                  scribe
+                  gif_creator
+                  polyvox
+                  artery-font-format
+                  cgltf
+                  etc2comp
+                  ;
+                inherit webrtc-audio-processing;
+                # FIXME: this doesn't work, bcs node18 uses icu75, which stops linking with overte
+                libv8 = pkgs.nodejs_18.libv8;
+              };
+            };
+            devShells.default = pkgs.mkShell {
+              packages = [
+                pkgs.conan
+                pkgs.ninja
+              ];
+              inputsFrom = [ self'.packages.overte ];
+              nativeBuildInputs = builtins.attrValues {
+                inherit (pkgs)
+                  cmake
+                  pkg-config
+                  python3
+                  ;
+              };
+              buildInputs =
+                builtins.attrValues {
+                  inherit (pkgs)
+                    openssl
+                    libGL
+                    glm
+                    nlohmann_json
+                    tbb_2022_0
+                    nodejs_18
+                    nvidia-texture-tools
+                    openexr
+                    draco
+                    bullet
+                    discord-rpc
+                    openvr
+                    openxr-loader
+                    SDL2
+                    libopus
+                    ;
+                  inherit (pkgs.libsForQt5)
+                    qtbase
+                    qtmultimedia
+                    qtdeclarative
+                    qtwebsockets
+                    qtsvg
+                    quazip
+                    ;
+                  inherit (pkgs.libsForQt5.qt5) qtwebchannel qtwebengine qtxmlpatterns;
+                  inherit (self'.packages)
+                    glad
+                    etc2comp
+                    cgltf
+                    polyvox
+                    gif_creator
+                    artery-font-format
+                    #libnode
+                    ;
+                  inherit webrtc-audio-processing;
+                }
+                ++ [
+                  pkgs.nodejs_18.libv8
+                ];
+              NVTT_DIR = "${pkgs.nvidia-texture-tools}";
+              GLSLANG_DIR = "${pkgs.glslang}/bin";
+              SCRIBE_DIR = "${self'.packages.scribe}/tools";
+              SPIRV_CROSS_DIR = "${pkgs.spirv-cross}/bin";
+              SPIRV_TOOLS_DIR = "${pkgs.spirv-tools}/bin";
+              CXXFLAGS = "-falign-functions";
```

I see you did your research.
We recently changed this to `-falign-functions=32`, but I don't think that matters.
There should probably be a comment for the next guy thought, since people usually don't expect function alignment to be necessary for a software to run properly.
```suggestion
              # Enable function alignment to fix crashes in relation to V8 scripting engine
              CXXFLAGS = "-falign-functions";
```